Tests/GitHubRepositories.tests.ps1

```diff
@@ -829,970 +741,1060 @@ try
+    Context "When specifying the 'TeamSlug' and 'OrganizationName' Parameters from the Pipeline" {
+        BeforeAll -ScriptBlock {
+            $getGitHubRepositoryTeamPermissionParms = @{
+                Uri = $repo.svn_url
             }
+            $repoPermission = $team |
+            Get-GitHubRepositoryTeamPermission @getGitHubRepositoryTeamPermissionParms
+        }
 
-            $team = New-GitHubTeam @newGithubTeamParms
+        It 'Should have the expected type and additional properties' {
+            $repoPermission.PSObject.TypeNames[0] | Should -Be $repositoryTeamPermissionTypeName
+            $repoPermission.RepositoryName | Should -Be $repo.full_name
+            $repoPermission.TeamName | Should -Be $teamName
+            $repoPermission.TeamSlug | Should -Be $team.TeamSlug
+        }
+    }
 
-            $permissions = 'Push', 'Pull', 'Maintain', 'Triage', 'Admin'
+    AfterAll {
+        if (Get-Variable -Name team -ErrorAction SilentlyContinue)
+        {
+            $team | Remove-GitHubTeam -Force
         }
 
-        Foreach ($permission in $permissions) {
-            Context "When the Team Permission is specified as $permission" {
-                BeforeAll {
-                    $setGitHubRepositoryTeamPermissionParms = @{
-                        Uri = $repo.svn_url
-                        TeamSlug = $team.slug
-                        Permission = $permission
-                    }
+        if (Get-Variable -Name repo -ErrorAction SilentlyContinue)
+        {
+            $repo | Remove-GitHubRepository -Force
+        }
+    }
+}
 
-                }
+Describe 'GitHubRepositories\Set-GitHubRepositoryTeamPermission' {
+    BeforeAll {
+        $repoName = [Guid]::NewGuid().Guid
+        $repo = New-GitHubRepository -OrganizationName $script:organizationName -RepositoryName $repoName
 
-                It 'Should not throw' {
-                    { Set-GitHubRepositoryTeamPermission @setGitHubRepositoryTeamPermissionParms }  |
-                        Should -Not -Throw
+        $teamName = [Guid]::NewGuid().Guid
+        $description = 'Team Description'
+        $privacy = 'closed'
+        $MaintainerName = $script:ownerName
 
-                }
+        $newGithubTeamParms = @{
+            OrganizationName = $script:organizationName
+            TeamName = $teamName
+            Description = $description
+            Privacy = $privacy
+            MaintainerName = $MaintainerName
+        }
 
-                It 'Should have set the correct Team permission' {
-                    $getGithubRepositoryTeamPermissionParms = @{
-                        Uri = $repo.svn_url
-                        TeamSlug = $team.slug
-                    }
+        $team = New-GitHubTeam @newGithubTeamParms
 
-                    $repoPermission = Get-GitHubRepositoryTeamPermission @getGithubRepositoryTeamPermissionParms
+        $permissions = 'Push', 'Pull', 'Maintain', 'Triage', 'Admin'
```

Comment:
```suggestion
```

Looks like we either drop that, or reference it at 1655 (I think dropping it is fine).